From 9baa841380561863e598ca7df64fc3cab11d673e Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Fri, 7 Jun 2019 07:43:24 +0200
Subject: [PATCH 1/1] efi_loader: RegisterProtocolNotify event signaling

In a following patch efi_signal_event() will only queue an event if it is
not signaled.

Set the is_signaled status to false before signaling the event.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_boottime.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 3072e85a0c..74aa5f9b7c 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1070,6 +1070,7 @@ efi_status_t efi_add_protocol(const efi_handle_t handle,
 			}
 			notif->handle = handle;
 			list_add_tail(&notif->link, &event->handles);
+			event->event->is_signaled = false;
 			efi_signal_event(event->event, true);
 		}
 	}
-- 
2.20.1

